docker/README.md

```diff
@@ -0,0 +1,4 @@
+This directory contains standard images maintained by Istio for test infrastructure purposes.
+
+* prowbazel - gives an environment with git, gcloud, python, bazel, and go. It is suitable for bazel-based projects.
```

Comment:
good point @rshriram -- the prowbazel docker file in particular, not tied to GCP, just a docker file (we do host it on GCR). 

to reiterate as @sebastienvas mentioned, the entrypoint of the docker file uploads logs to a gcs bucket, which is google specific. the bucket is read public.

other folks running infrastructure would likely use an image that uploads logs to their preferred storage.